### ISSUE TITLE
Support more Markdown formating 

### ISSUE THREAD

**Issue by molikto** [WuTheFWasThat/vimflowy#276]
* formats
    * `code blocks`
    * multi paragraph item
       like this
    * numbered list
          1. item 1
         2. item 2
    * ```
      raw code
       ```

**Comment by WuTheFWasThat** [comment]
multiparagraph should work with shift+enter in insert mode

it may not be too hard to support these things - see src/plugins/html/index.tsx and src/plugins/latex/index.tsx for examples of similar plugins.  you can already do those things via the html plugin, though html is definitely less nice than markdown!

**Comment by molikto** [comment]
After sometime trying out, I think the parser should directly support formating, not as a plugin. For example, I am migrating some old notes from Pandoc, and **$\omega$-accumulation point** is not displayed correctly. 

I might write my own version of vimflowy using Kotlin and Kotlin.js... let's see -_-

**Comment by WuTheFWasThat** [comment]
yeah, I knew that having it as a plugin creates some limitations/issues, but I wanted it that way for other reasons.  

whoah, cool!  let me know if you do, I'd be interested in looking at the code and trying it out!

**Comment by jcjolley** [comment]
@WuTheFWasThat he did it! https://github.com/molikto/CherryTree

**Comment by WuTheFWasThat** [comment]
yeah, I tried an early version at some point!  @molikto how has it been going?

**Comment by molikto** [comment]
I've been using it personally and satisfied with it, but I am no longer activity developing it anymore:

* It is not a good suit for a commercial product: it is a very small market, maybe just me will need something like this. One of Vimflowy, Dynalist or plain Markdown will be preferred by most other people
* I cannot get some fundamental idea right: I got distracted trying to create a node-level styling system such that can express logical structures of a document in a compossible way, but found problems like [this (search "logical" paragraphs)](https://html.spec.whatwg.org/multipage/grouping-content.html#the-p-element)

**Comment by jeanlucthumm** [comment]
I would also really like the ability to handle mark down style web links. I don't think there's a way to shorten links at this point so I end up with just big ass URLs in my notes

**Comment by WuTheFWasThat** [comment]
there's an HTML plugin so that you can just inline raw html!
